src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -152,16 +143,20 @@ private string GetQualifiedTypeName(ITypeSymbol typeInfo)
                     {
                         return $"System.Tuple<{string.Join(", ", tupleUnderlyingType.TypeArguments.Select(x => x.ToString()))}>";
                     }
+
                     return typeInfo.ToString();
                 }
 
-                return typeInfo.ContainingNamespace?.ToString() + "." + typeInfo.Name?.ToString();
+                if (!typeInfo.Name.Equals(""))
+                {
+                    return typeInfo.ContainingNamespace?.ToString() + "." + typeInfo.Name?.ToString();
```

Comment:
the `?` in the`typeInfo.Name?.ToString()` expression is not needed since you already validated that `typeInfo.Name` is not null on line 150.